### PR TITLE
Conveniences

### PR DESCRIPTION
**Pull request by rafiki270**
- giving optional column keys default value
- adding support for "column subqueries" as `DataComputedColumn` is unusable for that

### CODE REVIEW

File: Sources/SQL/Data/DataSubqueryColumn.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/// A computed SQL column.
+public struct DataSubqueryColumn {
+    /// The SQL sub-query
+    public var query: String
```
Comment (tanner0101) [inline Sources/SQL/Data/DataSubqueryColumn.swift]: Subquery should be a query struct, not just a string. The serializer should take care of serializing it.

**Comment by rafiki270** [comment]
@tanner0101 done

File: Tests/SQLTests/DataQueryTests.swift [hunk 2]
```diff
@@ -2,6 +2,21 @@ import SQL
 import XCTest
 
 final class DataQueryTests: XCTestCase {
+    
+    static let allTests = [
```
Comment (tanner0101) [inline Tests/SQLTests/DataQueryTests.swift]: Can you move this to the bottom of the file? That will match the rest of the tests in Vapor packages.

Comment (rafiki270) [reply Tests/SQLTests/DataQueryTests.swift]: I was actually hoping you would agree to have it on the top ... it's just too easy to forget to add the tests for linux and without testing we got all tests I hoped having it on the top always would help?

File: Sources/SQL/DataQuery/DataQueryColumn.swift [hunk 3]
```diff
@@ -1,11 +1,44 @@
...
+    let storage: Storage
+    
+    /// Internal initializer
+    init(_ stored: Storage) {
+        storage = stored
+    }
+    
     /// All columns, `*`.
-    case all
+    public static var all = DataQueryColumn(.all)
```
Comment (MrMage) [inline Sources/SQL/DataQuery/DataQueryColumn.swift]: Should this be a `static let`?

File: Sources/SQL/Serialization/SQLSerializer+DataSubquery.swift [hunk 4]
```diff
@@ -0,0 +1,8 @@
+extension SQLSerializer {
+    /// See `SQLSerializer`.
+    public func serialize(column: DataSubqueryColumn) -> String {
+        let query = serialize(query: column.query)
+        let serialized = "(\(query))"
```
Comment (MrMage) [inline Sources/SQL/Serialization/SQLSerializer+DataSubquery.swift]: Super-nit: I've heard that string interpolation can be much slower than concatenation (see https://mjtsai.com/blog/2018/03/09/faster-swift-string-concatenation/). Would it be worth it to change to `"(" + query + ")"` (depends on how often this going to be called, of course)?

File: Sources/SQL/Data/DataSubqueryColumn.swift [hunk 5]
```diff
@@ -0,0 +1,12 @@
+/// A computed SQL column.
+public struct DataSubqueryColumn {
+    /// The SQL sub-query
+    public var query: DataQuery
```
Comment (MrMage) [inline Sources/SQL/Data/DataSubqueryColumn.swift]: `let`?

**Pull request closed without merging**